packages/contracts-periphery/contracts/universal/citizen-house/BadgeAdmin.sol

```diff
@@ -0,0 +1,622 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import { IBadge } from "./IBadge.sol";
+
+/**
+ * @title  Badge Administrator
+ * @notice The Badge Administrator is intended to handle the adminstration of Citizen House and its
+ *         periphery contracts. The control is heirarchial and consists of three seperate roles:
+ *         OP, OP Company, and Citizen. The OP role grants top level access, with the ability to
+ *         add/block OP Companies, as well as add other OPs. The OP Company role grants secondary
+ *         access with the ability to add/block/remove OP Company Citizens. The Citizen role grants
+ *         access to the Citizen House and its periphery contracts, namely the ability to mint a
+ *         Soulbound ERC-721 token which is used to participate by vote in a Citizen House grant
+ *         round.
+ * @author OPTIMISM + GITCOIN
+ */
+contract BadgeAdmin is Ownable {
+    /**
+     * @notice Represents an OP.
+     */
+    struct OP {
+        address op;
+        bytes32 metadata;
+    }
+
+    /**
+     * @notice Represents an OP Company.
+     */
+    struct OPCO {
+        address co;
+        bool valid;
+        address[] citizens;
+        uint256 supply;
+        uint256 minted;
+        bytes32 metadata;
+    }
+
+    /**
+     * @notice Represents a Citizen.
+     */
+    struct Citizen {
+        address citizen;
+        bool valid;
+        bool minted;
+        address opco;
+        bytes ballot;
+        address delegate;
+        uint256 power;
+        bytes32 metadata;
+    }
+
+    /**
+     * @notice Maximum number of OP roles that can assigned in a single tx.
+     */
+    uint256 public maxOPLimit;
+
+    /**
+     * @notice Maximum number of OP company roles that can be assigned in a single tx.
+     */
+    uint256 public maxOPCOLimit;
+
+    /**
+     * @notice Maximum number of Citizen roles that can be assigned in a single tx.
+     */
+    uint256 public maxCitizenLimit;
```

Comment:
If it is intentional that this is not able to be changed, it can be an immutable